src/SqlClient/SqlCommandProvider.fs

```diff
@@ -109,41 +109,51 @@ type SqlCommandProvider(config : TypeProviderConfig) as this =
         conn.CheckVersion()
         conn.LoadDataTypesMap()
 
-        let designTimeSqlStatement, tempTableTypes = 
-            let sql, types = DesignTime.SubstituteTempTables(conn, sqlStatement, tempTableDefinitions)
-            DesignTime.SubstituteTableVar(sql, tableVarMapping), types
+        let connectionId = Guid.NewGuid().ToString().Substring(0, 8)
```

Comment:
Can we have explanation on why we need this? it seems to be used for the provided type name of the table, maybe better to pick something deterministic, or to remove it from here and internal only to the place it is used.

What would be the problem to just use the whole connection string instead?

Reply:
or maybe the GetHashCode on the connection string, this way we can probably avoid passing a parameter and keep it as implementation detail of the substitution code.

Reply:
In some cases parallel msbuild or msbuild and visual studio can both be building the type provider at the same time so you end up with both trying to create, use and drop the same temp table which creates build errors.  Adding a Guid in there stops them from fighting.